Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -173,9 +179,32 @@ - (NSData *)measureNSDataFromURL:(NSURL *)url
 
     [ioSpan setDataValue:path forKey:@"file.path"];
 
+    [self mainThreadExtraInfo:ioSpan];
+
     return ioSpan;
 }
 
+- (void)mainThreadExtraInfo:(id<SentrySpan>)span
+{
+    BOOL isMainThread = [NSThread isMainThread];
+
+    [span setDataValue:@(isMainThread) forKey:@"blocked_main_thread"];
+
+    if (!isMainThread) {
+        return;
+    }
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
```

Comment:
`m`: I would prefer using the `SentryDependencyContainer` or even better pass in the `SentryThreadInspector` as a dependency to `SentryNSDataTracker`, remove the `sharedInstance` from `SentryNSDataTracker` and add the `SentryNSDataTracker` to the `SentryDependencyContainer`, and `SentryNSDataSwizzling` uses the `SentryDependencyContainer` instead of the `SentryNSDataTracker.sharedInstance`.